### ISSUE TITLE
the configuration file path is all over the place

### ISSUE THREAD

**Issue by dhrp** [open-telemetry/opentelemetry-collector-contrib#31403]
### Component(s)

cmd/opampsupervisor, cmd/otelcontribcol, exporter/googlecloud

## Description

The problem is that it seems that there are several "standard" locations to put the config file. I have seen:

/etc/otel/config.yaml
/etc/otelcol/config.yaml
/ec/otelcol-contrib/config.yaml

And there are several Docker Image releases that read it, by default, from different places. - As a result getting started with OpenTelemetry is harder than it needs to be. I wasted time following this example on the googelcloudexporter, only to later find out the config path in the example is wrong. 

docker run \
  --volume ~/service-account-key.json:/etc/otel/key.json \
  --volume $(pwd)/config.yaml:/etc/otel/config.yaml \
  --env GOOGLE_APPLICATION_CREDENTIALS=/etc/otel/key.json \
  --expose 4317 \
  --expose 55681 \
  --rm \
  otel/opentelemetry-collector-contrib

I'll make a PR to fix this instance on googelcloudexporter but I see it is happening on a lot of places: Amongst others: (search in the repo for "/etc/ote"

https://github.com/dhrp/opentelemetry-collector-contrib/blob/main/cmd/opampsupervisor/specification/README.md#L137
https://github.com/dhrp/opentelemetry-collector-contrib/blob/main/cmd/otelcontribcol/Dockerfile#L15
https://github.com/dhrp/opentelemetry-collector-contrib/blob/3652b444802fad4819b50fbda41e0957000d8279/examples/couchbase/docker-compose.yaml#L16
https://github.com/dhrp/opentelemetry-collector-contrib/blob/main/exporter/googlecloudexporter/README.md#L60

What I propose is picking one location, and unifying all Dockerfiles, documentation and stuff to use that path by default. I know it's probably not easy to do, as it "may" break things, depending on how it's done, it'll save lots of time in the future.

## Steps to Reproduce

Follow the instructions on https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/googlecloudexporter/README.md, add the provided config file onto the suggested path. 

## Expected Result

Logs flowign to Google Cloud

## Actual Result

Some debug logs coming out of the container, but nothing showing up on google. - No errors. -> Didn't realize config was never used!

### Collector version

v0.95.0

## Environment
All

### OpenTelemetry Collector configuration

```yaml
-
```

**Comment by astencel-sumo** [comment]
I agree it would be easier if there was a standard configuration path used by Otelcol Docker images, at the very least by the Otelcol core and contrib.

It would also help if the Overview for Otel [core](https://hub.docker.com/r/otel/opentelemetry-collector) and [contrib](https://hub.docker.com/r/otel/opentelemetry-collector-contrib/) in Docker Hub was not empty, but instead described how to use the images with your own configuration.

![image](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/70892616/8a39de14-5460-42da-b9f5-7affd9d4c7ad)

**Comment by crobert-1** [comment]
My main concern is how big of a breaking change this may be, but overall it's a good idea. It's confused me plenty of times, so I agree it would be nice to have one path. Feel free to submit a PR and the impact and path can be discussed there.

**Comment by mx-psi** [comment]
Probably the minority opinion here, but I feel like having a default configuration in Dockerfiles is not very useful and if we were to change it my vote would be to remove the default configuration entirely

**Comment by astencel-sumo** [comment]
@crobert-1 has a good point that changing the paths would be a breaking change of big blast radius. I'd be fine with just documenting the paths used currently in the image registry (Docker Hub).

**Comment by mx-psi** [comment]
This was recently brought up again on open-telemetry/opentelemetry-collector-releases/issues/530. I am going to close this and we can continue discussing on the other issue, since this pertains to opentelemetry-collector-releases
